src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -50,9 +50,18 @@
     </PropertyGroup>
 
     <PropertyGroup>
-      <LangVersion Condition="'$(LangVersion)' == '' AND
-        (('$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(TargetFrameworkVersion)' == 'v3.0') OR
-         ('$(TargetFrameworkIdentifier)' == '.NETStandard' AND '$(TargetFrameworkVersion)' == 'v2.1'))">preview</LangVersion>
+      <!-- To aid certain IDE features we make a distinction between the maximum supported language version, and the version 
+           actually used during compilation. .NETCore 3.0 supports up to C# 8.0, while all others support up to 7.3 by default.
+           Note that other frameworks (e.g., Xamarin, Unity, etc.) are expected to override this by setting
+           MaxSupportedLangVersion themselves. -->
+      <!-- Strip the 'v' off the beginning of TargetFrameworkVersion so we can actually compare it as a version. -->
+      <_TargetFrameworkVersionAsVersion>$(TargetFrameworkVersion.Substring(1))</_TargetFrameworkVersionAsVersion>
```

Comment:
We have _TargetFrameworkVersionWithoutV defined by SDK, and despite the fact that it is "private" by underscore convention, others use it.

Elsewhere, we are more defensive and do TrimStart(vV) instead of Substring(1). 

I think maybe instead of creating yet another private variable people may depend upon, you could consider inlining it like: https://github.com/microsoft/msbuild/blob/0cd0e92a7243088977d31b56626b56d6116de016/src/Tasks/Microsoft.Common.CurrentVersion.targets#L328

Longer term, I would want to allow leading v as a valid version as mentioned here: https://github.com/microsoft/msbuild/issues/3212#issuecomment-413954918

Reply:
I guess inlining wouldn't be good for perf when there is more than one version below.

Reply:
cc @rainersigwald for thoughts here.

Reply:
I lean toward "use the SDK `_TargetFrameworkVersionWithoutV`", or we could use this as an excuse to implement `$([MSBuild]::CompareVersions())` (but the coordinated insertion would be a pain; do it like this initially either way).

Reply:
I can't count on `_TargetFrameworkVersionWithoutV` having been defined, though.

Reply:
oh, sure, this has to work outside the sdk. Then I'd switch to trimstart.

Reply:
Consider clearing _TargetFrameworkAsVersion too. I really want to avoid more turds we get scared to clean up after we put in a proper feature in msbuild.

Reply:
That's fair. Is there a canonical way of doing that? Will `<_TargetFrameworkVersionAsVersion></_TargetFrameworkVersionAsVersion>` do it? I would _assume_ so, but assumptions are dangerous when it comes to MSBuild. :-)

Reply:
Yes, or the self-closing tag version `<_TargetFrameworkVersionAsVersion />`. Both "set it to empty" instead of "deleting the variable", but that's basically indistinguishable. 

But in this case can you just inline it? I only see one use . . .

Reply:
Inlining was my first suggestion, but there will be another version check in here eventually. Maybe better to have the code ready to evolve. When someone goes to add a second one, they may not catch what we've discussed here...